doc/building.rst

```diff
@@ -15,88 +15,151 @@ These instructions are updated to work with version |version| of the Nextcloud C
 
-4. Add the Nextcloud repository using the following command::
+.. _`as recommended by Apple`: https://support.apple.com/en-us/HT208050
 
-    brew tap owncloud/owncloud
+7. Install the Packages package creation tool:
 
-5. Install a Qt5 version with qtwebkit support::
+   http://s.sudre.free.fr/Software/Packages/about.html
 
-    brew install qt5
+8. Clone the Nextcloud repository to a convenient location, such as ``~/Repositories``:
 
-6. Install any missing dependencies::
+   .. code-block:: bash
 
-    brew install $(brew deps owncloud-client)
+      % cd ~/Repositories
 
-7. Add Qt from brew to the path::
+   .. note:: The cloned repository can go basically anywhere your user account
+      has write access, though it should not go in a directory synced with another
+      cloud service (especially not iCloud Drive). ``~/Repositories`` is recommended
+      for tidiness and consistency.
 
-    export PATH=/usr/local/Cellar/qt5/5.x.y/bin:$PATH
+   Then, assuming you've already forked Nextcloud Desktop:
 
-   Where ``x.y`` is the current version of Qt 5 that brew has installed
-   on your machine.
-8. Install qtkeychain from here:  git clone https://github.com/frankosterfeld/qtkeychain.git
-   make sure you make the same install prefix as later while building the client e.g.  -
-   ``DCMAKE_INSTALL_PREFIX=/Path/to/client-install``
+   .. code-block:: bash
 
-9. For compilation of the client, follow the :ref:`generic-build-instructions`.
+      % git clone git@github.com:<github_username>/desktop.git
 
-10. Install the Packages_ package creation tool.
+   Or, for HTTPS (soon to be deprecated):
 
-11. Enable git submodules:
-```
-$ cd desktop
-$ git submodule init
-$ git submodule update
-```
+   .. code-block:: bash
 
-12. Generate the build files:
-```
-$ cd build
-$ cmake .. -DCMAKE_INSTALL_PREFIX=~/nextcloud-desktop-client -DCMAKE_BUILD_TYPE=Debug
-```
+      % git clone https://github.com/<github_username>/desktop.git
 
-13. Compile and install:
-```
-$ make install
-```
+   Or, if you haven't forked Nextcloud Desktop, clone as read-only:
 
-   .. note:: Contrary to earlier versions, Nextcloud 1.7 and later are packaged
-             as a ``pkg`` installer. Do not call "make package" at any time when
-             compiling for OS X, as this will build a disk image, and will not
-             work correctly.
+   .. code-block:: bash
+
+      % git clone git@github.com:nextcloud/desktop.git
+
+   Or, for HTTPS (soon to be deprecated):
+
+   .. code-block:: bash
+
+      % git clone https://github.com/nextcloud/desktop.git
+
+9. Enable git submodules:
+
+   .. code-block:: bash
+
+      % cd ~/Repositories/desktop
+      % git submodule update --init
+
+12. Create build directory:
+
+   .. code-block:: bash
+
+      % cd ~/Repositories/desktop
+      % mkdir build
+
+13. Generate the build files:
+
+   .. code-block:: bash
+
+      % cd ~/Repositories/desktop/build
+      % cmake .. \
+         -DCMAKE_INSTALL_PREFIX=~/Builds
```

Comment:
I also needed to set `-D OPENSSL_CRYPTO_LIBRARY=$(brew --prefix openssl)/lib/libcrypto.dylib -D OPENSSL_SSL_LIBRARY=$(brew --prefix openssl)/lib/libssl.dylib`. This was necessary because it seems like that CMake does detect the system libraries first but we can not use them because they are not really OpenSSL libraries but LibreSSL libraries. They just happen to have the same names. From my understanding that is a problem in CMake's `FindOpenSSL.cmake`. But maybe someone knows it better than me;)

Reply:
If you set the `.zshrc` the way I specified in this version of instructions, this is not a problem. If you specify the variables some other way, it might very well be. I tested the instructions in #2983 multiple times. I can change this if it becomes necessary with other changes to the instructions.

Reply:
Interestingly I rerun my build without specifying them and there were no errors. Maybe I had a typo the last time. I guess we then don't need to add that.